Userland/Libraries/LibJS/Tests/builtins/Temporal/Duration/Duration.prototype.round.js

```diff
@@ -0,0 +1,164 @@
+describe("correct behavior", () => {
+    test("length is 1", () => {
+        expect(Temporal.Duration.prototype.round).toHaveLength(1);
+    });
+
+    test("basic functionality", () => {
+        const duration = new Temporal.Duration(0, 0, 0, 21, 7, 10, 100, 200, 300, 400);
+        const values = [
+            ["nanosecond", "P21DT7H11M40.2003004S"],
+            ["microsecond", "P21DT7H11M40.2003S"],
+            ["millisecond", "P21DT7H11M40.2S"],
+            ["second", "P21DT7H11M40S"],
+            ["minute", "P21DT7H12M"],
+            ["hour", "P21DT7H"],
+            ["day", "P21D"],
+        ];
+
+        for (const [smallestUnit, durationString] of values) {
+            const singularRoundedDuration = duration.round({ smallestUnit });
+            const pluralRoundedDuration = duration.round({ smallestUnit: `${smallestUnit}s` });
+
+            // Passing in a string is treated as though { smallestUnit: "<string value>" } was passed in.
+            const singularRoundedDurationWithString = duration.round(smallestUnit);
+            const pluralRoundedDurationWithString = duration.round(`${smallestUnit}s`);
+
+            expect(singularRoundedDuration.toString()).toBe(durationString);
+            expect(singularRoundedDurationWithString.toString()).toBe(durationString);
+            expect(pluralRoundedDuration.toString()).toBe(durationString);
+            expect(pluralRoundedDurationWithString.toString()).toBe(durationString);
+        }
+    });
+
+    test("largestUnit option", () => {
+        const duration = new Temporal.Duration(0, 0, 0, 21, 7, 10, 100, 200, 300, 400);
+
+        // Using strings is not sufficient here, for example, the nanosecond case will produce "PT1840300.2003004S" which is 1840300 s, 200 ms, 300 us, 400 ns
```

Comment:
Good call, I think I made this mistake in one other tests (which isn't *wrong*, but not fully testing the implementation)